### PR TITLE
Dropping support for symfony < 3.4

### PR DESCRIPTION
**Pull request by franmomu**
## Subject

This is the same thing as https://github.com/sonata-project/SonataAdminBundle/pull/5733, dropping support of Symfony < 3.4 and >= 4, < 4.2

I am targeting this branch, because these changes respect BC.

## Changelog

```markdown
### Removed
- Support for Symfony < 3.4
- Support for Symfony >= 4, < 4.2
```

### CODE REVIEW

File: docs/reference/form_field_definition.rst [hunk 1]
```diff
@@ -139,15 +141,17 @@ If you have many ``Post`` linked to one ``User``, then the ``Post`` form should
 
 The AdminBundle provides 2 options:
 
-* ``sonata_type_model``: the ``User`` list is set in a select widget with an `Add` button to create a new ``User``,
-* ``sonata_type_model_list``: the ``User`` list is set in a model where you can search, select and delete a ``User``.
+* ``\Sonata\AdminBundle\Form\Type\ModelType``: the ``User`` list is set in a select widget with an `Add` button to create a new ``User``,
```
Comment (OskarStark) [inline docs/reference/form_field_definition.rst]:
```suggestion
* ``Sonata\AdminBundle\Form\Type\ModelType``: the ``User`` list is set in a select widget with an `Add` button to create a new ``User``,
```

File: docs/reference/form_field_definition.rst [hunk 2]
```diff
@@ -139,15 +141,17 @@ If you have many ``Post`` linked to one ``User``, then the ``Post`` form should
...
+* ``\Sonata\AdminBundle\Form\Type\ModelListType``: the ``User`` list is set in a model where you can search, select and delete a ``User``.
```
Comment (OskarStark) [inline docs/reference/form_field_definition.rst]:
```suggestion
* ``Sonata\AdminBundle\Form\Type\ModelListType``: the ``User`` list is set in a model where you can search, select and delete a ``User``.
```

File: docs/reference/form_types_and_transformers.rst [hunk 3]
```diff
@@ -10,13 +10,13 @@ The `AdminBundle` is shipped with custom form types and data transfomers in orde
 Form types
 ----------
 
-* ``sonata_type_admin``: this type is linked to an `Admin` class and the field construction is delegated to an Admin class,
-* ``sonata_type_collection``: this type works like the native ``CollectionType`` but contains two extra features:
+* ``\Sonata\AdminBundle\Form\Type\AdminType``: this type is linked to an `Admin` class and the field construction is delegated to an Admin class,
```
Comment (OskarStark) [inline docs/reference/form_types_and_transformers.rst]:
```suggestion
* ``Sonata\AdminBundle\Form\Type\AdminType``: this type is linked to an `Admin` class and the field construction is delegated to an Admin class,
```

File: docs/reference/form_types_and_transformers.rst [hunk 4]
```diff
@@ -10,13 +10,13 @@ The `AdminBundle` is shipped with custom form types and data transfomers in orde
...
+* ``\Sonata\Form\Type\CollectionType``: this type works like the native ``CollectionType`` but contains two extra features:
```
Comment (OskarStark) [inline docs/reference/form_types_and_transformers.rst]:
```suggestion
* ``Sonata\Form\Type\CollectionType``: this type works like the native ``CollectionType`` but contains two extra features:
```

File: docs/reference/form_types_and_transformers.rst [hunk 5]
```diff
@@ -10,13 +10,13 @@ The `AdminBundle` is shipped with custom form types and data transfomers in orde
...
 
     * the data layer is abstracted to work with any implemented layer,
     * a delete option is added so a collection entry can be deleted.
-* ``sonata_type_model``: this type works like the native ``EntityType`` but this internal is abstracted to work with any implemented layer.
-* ``sonata_type_immutable_array``: this type allows to edit a fixed array, like a settings array.
+* ``\Sonata\AdminBundle\Form\Type\ModelType``: this type works like the native ``EntityType`` but this internal is abstracted to work with any implemented layer.
```
Comment (OskarStark) [inline docs/reference/form_types_and_transformers.rst]:
```suggestion
* ``Sonata\AdminBundle\Form\Type\ModelType``: this type works like the native ``EntityType`` but this internal is abstracted to work with any implemented layer.
```

File: docs/reference/form_types_and_transformers.rst [hunk 6]
```diff
@@ -10,13 +10,13 @@ The `AdminBundle` is shipped with custom form types and data transfomers in orde
...
+* ``\Sonata\Form\Type\ImmutableArrayType``: this type allows to edit a fixed array, like a settings array.
```
Comment (OskarStark) [inline docs/reference/form_types_and_transformers.rst]:
```suggestion
* ``Sonata\Form\Type\ImmutableArrayType``: this type allows to edit a fixed array, like a settings array.
```

File: src/Builder/FormContractor.php [hunk 7]
```diff
@@ -114,16 +108,15 @@ public function getDefaultOptions($type, FieldDescriptionInterface $fieldDescrip
         ])) {
             if ('list' === $fieldDescription->getOption('edit')) {
                 throw new \LogicException(
-                    'The `sonata_type_model` type does not accept an `edit` option anymore,'
+                    'The `\Sonata\AdminBundle\Form\Type\ModelType` type does not accept an `edit` option anymore,'
```
Comment (OskarStark) [inline src/Builder/FormContractor.php]:
```suggestion
                    'The `Sonata\AdminBundle\Form\Type\ModelType` type does not accept an `edit` option anymore,'
```

File: src/Builder/FormContractor.php [hunk 8]
```diff
@@ -114,16 +108,15 @@ public function getDefaultOptions($type, FieldDescriptionInterface $fieldDescrip
         ])) {
             if ('list' === $fieldDescription->getOption('edit')) {
                 throw new \LogicException(
-                    'The `sonata_type_model` type does not accept an `edit` option anymore,'
+                    'The `Sonata\AdminBundle\Form\Type\ModelType` type does not accept an `edit` option anymore,'
```
Comment (core23) [inline src/Builder/FormContractor.php]: Can you please use `sprintf` here and use `ModelType::class`

File: src/Builder/FormContractor.php [hunk 9]
```diff
@@ -149,8 +141,8 @@ public function getDefaultOptions($type, FieldDescriptionInterface $fieldDescrip
                 ClassMetadata::MANY_TO_ONE,
             ], true)) {
                 throw new \RuntimeException(sprintf(
-                    'You are trying to add `sonata_type_admin` field `%s` which is not One-To-One or  Many-To-One.'
-                    .' Maybe you want `sonata_type_collection` instead?',
+                    'You are trying to add `Sonata\AdminBundle\Form\Type\AdminType` field `%s` which is not One-To-One or  Many-To-One.'
```
Comment (core23) [inline src/Builder/FormContractor.php]: Can you please use `sprintf` here and use `ModelType::class`

File: tests/Datagrid/OrderByToSelectWalkerTest.php [hunk 10]
```diff
@@ -55,10 +54,7 @@ public function testOrderByCompositeId(): void
         $query->setHint(Query::HINT_CUSTOM_TREE_WALKERS, [OrderByToSelectWalker::class]);
 
         $this->assertSame(
-            // NEXT_MAJOR: Remove this check when dropping support for doctrine/orm < 2.5
-            version_compare(Version::VERSION, '2.5') < 0
```
Comment (core23) [inline tests/Datagrid/OrderByToSelectWalkerTest.php]: We are dropping doctrine 2.5 too?

Comment (franmomu) [reply tests/Datagrid/OrderByToSelectWalkerTest.php]: In composer.json we have:

```
"doctrine/orm": "^2.5",
```

File: docs/reference/form_types_and_transformers.rst [hunk 11]
```diff
@@ -97,4 +99,4 @@ DataTransformer
 
 * ``ArrayToModelTransformer``: transform an array to an object,
 * ``ModelsToArrayTransformer``: transform a collection of array into a collection of object,
-* ``ModelToIdTransformater``: transform an ``id`` into an object.
\ No newline at end of file
+* ``ModelToIdTransformater``: transform an ``id`` into an object.
```
Comment (greg0ire) [inline docs/reference/form_types_and_transformers.rst]: I don't know what changed here, but shouldn't it be 
```suggestion
* ``ModelToIdTransformer``: transform an ``id`` into an object.
```

**Comment by OskarStark** [comment]
Great work 😊👍🏻
Thanks

**Pull request merged**